rebalance/rebalance.py

```diff
@@ -620,16 +633,68 @@ def maybe_rebalance_pairs(ch1, ch2, failed_channels: list):
     return result
 
 
+def rebalance_pair_picker(threadid, channel_pairs: list, failed_channels: list):
+    plugin.threadids[threading.get_ident()] = threadid
+    result = {"success": False, "fee_spent": Millisatoshi(0)}
+    idle_count = 0
+    while not rebalance_stopping():
+        idle_count += 1
+        for pair in channel_pairs:
+            if rebalance_stopping():
+                return result
+            ch1 = pair[0]
+            ch2 = pair[1]
+            processed = pair[2]
+            if processed:
+                continue
+            if ch1["lock"].acquire(blocking=False):
+                if ch2["lock"].acquire(blocking=False):
+                    idle_count = 0
+                    pair[2] = True
+                    result = maybe_rebalance_pairs(ch1, ch2, failed_channels)
```

Comment:
Understanding: Is it correct that the result does not need to get add-up here, as only successful results will be handled by the thread pool execution futures (further below) anyway and the unsuccessful runs do not need to be accounted for?

Reply:
Yes, we cannot do anything with the unsuccessful runs. It's the same as before.